### PR TITLE
For closed issue #5613. Add test to ensure spatial_select to dataset.transform from non-zero indexed dataframe is applied to right indexes 

### PR DESCRIPTION
**Pull request by GeoVizNow**
I added a test for this as it recently failed in 1.15.2 as described in issue 5613. Then, when running the updated 1.15.4, I realized there was not a problem in that version. So probably accounted for in 1.15.3 or 1.15.4.

Even though I am uncertain if there is any test to fetch this problem if reappearing and hence the following test. I realize I do not follow the setup with tests in classes, but hope you will consider the content of the test anyhow. And potentially you can recommended how you would like it modified.

### CODE REVIEW

**Comment by GeoVizNow** [comment]
By the way I should have mentioned one experience: I had troubles getting the command 
`doit develop_install -c pyviz/label/dev -c conda-forge -o tests -o examples`
to run when trying to make an editable install. Eventually I found (by googling) that if pinning setuptools to version '63.4.3' in the 'pyproject.toml', then the editable install went fine.

File: holoviews/tests/util/test_transform.py [hunk 1]
```diff
@@ -4,8 +4,13 @@
 import pickle
 import warnings
 
+import holoviews as hv
+from io import StringIO
+hv.extension('bokeh')
```
Comment (Hoxbro) [inline holoviews/tests/util/test_transform.py:9]: `hv.extension` should not be run in the test suite, and it isn't needed for your test to pass.

File: holoviews/tests/util/test_transform.py [hunk 2]
```diff
@@ -520,3 +525,56 @@ def test_pickle(self):
...
+        101|   P1|  7.|   3.
+        102|   P2|  3.|   4.
+        103|   P3|  .5|   3.
+        104|   P4|  2.|   2.
+        105|   P5|  1.|   2.
+        106|   P6|  1.|   1.
+        """)
+
+    # read in data using non-zero based index
+    return pd.read_csv(StringIO(data_text), sep='|', skipinitialspace=True, index_col=0)
```
Comment (Hoxbro) [inline holoviews/tests/util/test_transform.py:550]: I think you have overengineered this. You could get a similar result by having `pd.DataFrame({"a": [7, 3, 0.5, 2, 1, 1], "b": [3, 4, 3, 2, 2, 1]}, index=range(101, 107))`.

File: holoviews/tests/util/test_transform.py [hunk 3]
```diff
@@ -520,3 +525,56 @@ def test_pickle(self):
...
+ 
+    
+
+def test_dataset_transform_by_spatial_select_expr_index_not_0_based(df_fixture):
```
Comment (Hoxbro) [inline holoviews/tests/util/test_transform.py:554]: The df_fixture should be removed, and create the DataFrame inside this test.

File: holoviews/tests/util/test_transform.py [hunk 4]
```diff
@@ -520,3 +525,56 @@ def test_pickle(self):
...
+    """Ensure 'spatial_select' expression works when index not zero-based.
+    
+    Use 'spatial_select' defined by four nodes to select index 104, 105.
+    Apply expression to dataset.transform to generate new 'flag' column where True
+    for the two indexes."""
+    arr = np.array(
```
Comment (Hoxbro) [inline holoviews/tests/util/test_transform.py:560]: This variable could be called geometry.

**Review by Hoxbro — CHANGES_REQUESTED** [review]
Thank you for the test. I have left some comments, but in general, it looks good. :slightly_smiling_face:

File: holoviews/tests/util/test_transform.py (hunk 1 above)
Comment (GeoVizNow) [reply holoviews/tests/util/test_transform.py:9]: Sounds good, I remove it.

File: holoviews/tests/util/test_transform.py (hunk 2 above)
Comment (GeoVizNow) [reply holoviews/tests/util/test_transform.py:550]: I could modify this if you prefer. I just started to appreciate to see tabular input in test where the arrangment matters.

File: holoviews/tests/util/test_transform.py (hunk 3 above)
Comment (GeoVizNow) [reply holoviews/tests/util/test_transform.py:554]: Ok, then I will remove the fixture.

File: holoviews/tests/util/test_transform.py (hunk 4 above)
Comment (GeoVizNow) [reply holoviews/tests/util/test_transform.py:560]: Thanks, I will rename it.

File: holoviews/tests/util/test_transform.py (hunk 2 above)
Comment (Hoxbro) [reply holoviews/tests/util/test_transform.py:550]: I would very much prefer it. It takes focus away from the actual test code.

Comment (GeoVizNow) [reply holoviews/tests/util/test_transform.py:550]: Then I sure will remove it.

**Comment by GeoVizNow** [comment]
> Thank you for the test. I have left some comments, but in general, it looks good. 🙂

Thank you for reviewing, appreciate your guidance here :-)

**Comment by GeoVizNow** [comment]
I think the test is adjusted to the better now.

**Comment by jbednar** [comment]
Thanks for adapting it!

**Review by Hoxbro — APPROVED** [review]

**Comment by Hoxbro** [comment]
> Thank you for reviewing, appreciate your guidance here :-)

No, thank you for taking the time to make this PR :+1:

**Pull request merged**